indexer/ftypes_matcher.cpp

```diff
@@ -419,33 +428,41 @@ string DebugPrint(HighwayClass const cls)
-      {HighwayClass::Trunk, c.GetTypeByPath({"highway", "trunk_link"})},
-      {HighwayClass::Trunk, c.GetTypeByPath({"route", "ferry"})},
-      {HighwayClass::Primary, c.GetTypeByPath({"highway", "primary"})},
-      {HighwayClass::Primary, c.GetTypeByPath({"highway", "primary_link"})},
-      {HighwayClass::Secondary, c.GetTypeByPath({"highway", "secondary"})},
-      {HighwayClass::Secondary, c.GetTypeByPath({"highway", "secondary_link"})},
-      {HighwayClass::Tertiary, c.GetTypeByPath({"highway", "tertiary"})},
-      {HighwayClass::Tertiary, c.GetTypeByPath({"highway", "tertiary_link"})},
-      {HighwayClass::LivingStreet, c.GetTypeByPath({"highway", "unclassified"})},
-      {HighwayClass::LivingStreet, c.GetTypeByPath({"highway", "residential"})},
-      {HighwayClass::LivingStreet, c.GetTypeByPath({"highway", "living_street"})},
-      {HighwayClass::Service, c.GetTypeByPath({"highway", "service"})},
-      {HighwayClass::Service, c.GetTypeByPath({"highway", "track"})}};
-  uint8_t const kTruncLevel = 2;
+
+  static map<uint32_t, HighwayClass> const kHighwayClasses = {
+      {c.GetTypeByPath({"highway", "motorway"}), HighwayClass::Trunk},
+      {c.GetTypeByPath({"highway", "motorway_link"}), HighwayClass::Trunk},
+      {c.GetTypeByPath({"highway", "trunk"}), HighwayClass::Trunk},
+      {c.GetTypeByPath({"highway", "trunk_link"}), HighwayClass::Trunk},
+      {c.GetTypeByPath({"route", "ferry"}), HighwayClass::Trunk},
+      {c.GetTypeByPath({"highway", "primary"}), HighwayClass::Primary},
+      {c.GetTypeByPath({"highway", "primary_link"}), HighwayClass::Primary},
+      {c.GetTypeByPath({"highway", "secondary"}), HighwayClass::Secondary},
+      {c.GetTypeByPath({"highway", "secondary_link"}), HighwayClass::Secondary},
+      {c.GetTypeByPath({"highway", "tertiary"}), HighwayClass::Tertiary},
+      {c.GetTypeByPath({"highway", "tertiary_link"}), HighwayClass::Tertiary},
+      {c.GetTypeByPath({"highway", "unclassified"}), HighwayClass::LivingStreet},
+      {c.GetTypeByPath({"highway", "residential"}), HighwayClass::LivingStreet},
+      {c.GetTypeByPath({"highway", "living_street"}), HighwayClass::LivingStreet},
+      {c.GetTypeByPath({"highway", "road"}), HighwayClass::LivingStreet},
+      {c.GetTypeByPath({"highway", "service"}), HighwayClass::Service},
+      {c.GetTypeByPath({"highway", "track"}), HighwayClass::Service},
+      {c.GetTypeByPath({"highway", "pedestrian"}), HighwayClass::Pedestrian},
+      {c.GetTypeByPath({"highway", "footway"}), HighwayClass::Pedestrian},
+      {c.GetTypeByPath({"highway", "bridleway"}), HighwayClass::Pedestrian},
+      {c.GetTypeByPath({"highway", "steps"}), HighwayClass::Pedestrian},
+      {c.GetTypeByPath({"highway", "cycleway"}), HighwayClass::Pedestrian},
+      {c.GetTypeByPath({"highway", "path"}), HighwayClass::Pedestrian},
+  };
+  uint8_t constexpr kTruncLevel = 2;
+  auto const highwayClassesEndIt = kHighwayClasses.cend();
```

Comment:
Do you think it's good to call method kHighwayClasses.cend() for every loop in "for (auto t : types)"?